oaknational/Oak-Web-Application
# [PUPIL-710] Add content guidance modal and message to sensitive lessons

## Description

Music year: 2010

- List of changes

Added new content guidance modal and text area to lessons that have content guidance attached.

## Ticket
https://www.notion.so/oaknationalacademy/OWA-Consent-for-isSensitive-lessons-2a03fa79ca56441ebd290a19e6e14763

## Design 
https://www.figma.com/design/zqpeecgzMGKRli2gUdJOLp/Pupil-Browse?node-id=5139-30785&m=dev

## How to test

1. Go to https://deploy-preview-2446--oak-web-application.netlify.thenational.academy/pupils/programmes/english-secondary-year-10-aqa/units/poetry-anthology-first-study-155/lessons/analysing-poppies/overview
2. You should see a contents guidance modal with the appropriate warnings.
3. You shouldn't be able to access any of the lesson without clicking "I understand, continue"
4. ESC and clicking the backdrop should not clear the modal.
5. The "Take me home" should take you back to the browse page

## Screenshots

| Current | New |
|----------|----------|
| <img width="1391" alt="image" src="https://github.com/oaknational/Oak-Web-Application/assets/587241/af366dfb-0c54-40c4-9dc0-b8c8bd23c2e8"> | <img width="1354" alt="image" src="https://github.com/oaknational/Oak-Web-Application/assets/587241/4bfdee1e-b804-4ca1-be7d-38c52aff46b8"> |

## Checklist
